crates/ruff_python_parser/src/lexer.rs

```diff
@@ -169,7 +170,11 @@ impl<'source> Lexer<'source> {
     }
 
     /// Lex an identifier. Also used for keywords and string/bytes literals with a prefix.
-    fn lex_identifier(&mut self, first: char) -> Result<Tok, LexicalError> {
+    fn lex_identifier(
+        &mut self,
+        first: char,
+        first_char_is_ascii: bool,
```

Comment:
Nit: I would call `first.is_ascii()` inside of the method. It avoids leaking the abstraction and the cost of determining if a character is ascii is a simple range comparision that might as well be cheaper than passing a value to a function call (requires pushing and poping from the stack).